lib/iris/tests/unit/analysis/test_Aggregator.py

```diff
@@ -156,6 +156,19 @@ def test_unmasked(self):
             self.assertArrayAlmostEqual(result, mock_return.copy())
         mock_method.assert_called_once_with(data, axis=axis)
 
+    def test_allmasked_1D_with_mdtol(self):
+        data = ma.masked_all((3,))
+        axis = 0
+        mdtol = 0.5
+        mock_return = ma.masked
+        with mock.patch.object(
```

Comment:
Iris.Test has a patch built in, so you can use `mock_method = self.patch.object(...` I think (slight benefit of reduced indenting)

Reply:
Thanks, I didn't know about this method.  However I can't seem to get it to work in this case.  From the docstring, the method seems to want to patch the class rather than the instance, so I tried
```python
mock_method = self.patch("iris.analysis.Aggregator.call_func", return_value=mock_return)
```
but got 
```
AttributeError: <class 'iris.analysis.Aggregator'> does not have the attribute 'call_func'
```
because `call_func` is added to the aggregator at instantiation.

I also tried patching the instance by
```python
mock_method = self.patch("self.TEST.call_func", return_value=mock_return)
```
but that gave me
```
ModuleNotFoundError: No module named 'self'
```

Reply:
Huh, turns out our implementation of `patch` doesn't have an `object` attribute, whereas `unittest.mock.patch` does. Seems good to leave this as it is then.